api/v1/api.go

```diff
@@ -210,3 +209,71 @@ func (api *API) respondError(w http.ResponseWriter, apiErr apiError, data interf
 		level.Error(api.logger).Log("msg", "failed to write data to connection", "err", err)
 	}
 }
+
+type metricJson map[string]interface{}
+
+func metricsToInterface(metrics []*dto.Metric, metricsType dto.MetricType) []metricJson {
+
+	jsonMetrics := make([]metricJson, len(metrics))
+
+	for i, m := range metrics {
+		individualMetric := make(metricJson)
+		if metricsType == dto.MetricType_SUMMARY {
```

Comment:
I know it's done the same in prom2json, but I think this would read better as a switch statement. (I'll also send you a PR for prom2json.)